### PR TITLE
Dtolmaco/web 229 make variables in serverblock explicit

### PR DESCRIPTION
**Pull request by dantol29**
# Added `Variables` struct for you to work with Config file! Spent the whole day!
↓
↓
Conf.hpp
image
↓
↓
ServerBlock.hpp
image
↓
↓
ServerBlock.hpp Variables struct
image

### CODE REVIEW

File: src/Config.cpp [hunk 1]
```diff
@@ -64,10 +64,8 @@ bool	Config::saveVariable(const std::string& line)
 
 	if (line[i++] != ';' || i < line.length()) // [;]
 		return (false);
-	// TODO: line[i + 1] != '\0'
 
-	_tmpServer.addVariable(key, value);
-	return (true);
+	return (_tmpServer.addVariable(key, value, false));
```
Comment (552020) [inline src/Config.cpp:68]: Can we have _tmpServer as _tmpBlockServer?

File: src/Config.cpp [hunk 2]
```diff
@@ -243,128 +210,99 @@ bool	Config::pathExists(std::map<std::string, std::string> list, std::string var
...
-				if (isVulnerablePath(it->second.substr(start, i - start)))
-					return (error("Config file: Path is vulnerable"));
-				return (true);
-			}
-		}
-		return (error("Config file: Invalid index"));
-	}
-	return (true);
-}
+// bool	Config::checkVariablesValue(Variables var)
```
Comment (552020) [inline src/Config.cpp:213]: Why so much uncommented code? Do we still need it?

File: src/ServerBlock.hpp [hunk 3]
```diff
@@ -3,6 +3,26 @@
 
 #include "webserv.hpp"
 
+// a list of all accepted variables
+// ---------------------------------
+// 1. listen, 2. server_name, 3. error_page,
+// 4. index, 5. root, 6. client_max_body_size, 7. autoindex, 
+// 8. allow_methods, 9. alias, (cgi_path, cgi_ext) - optional
+// ---------------------------------
+struct Variables
```
Comment (552020) [inline src/ServerBlock.hpp:12]: Can we change Variables to Directives?

File: src/ServerBlock.hpp [hunk 4]
```diff
@@ -11,24 +31,45 @@ class ServerBlock
 		ServerBlock& operator=(const ServerBlock& obj);
 		~ServerBlock();
 
-		std::map<std::string, std::string> getVariables() const; // variables outside of locations
-		std::pair<std::string, std::string> getVariables(std::string key) const;
-		std::vector<std::map<std::string, std::string> > getLocations() const; // location / {} blocks
-		
-		void addVariable(std::string& key, std::string& value);
-		void addLocation(std::map<std::string, std::string>& var);
+		bool addVariable(std::string key, std::string& value, bool isLocation);
```
Comment (552020) [inline src/ServerBlock.hpp:34]: Is addVariable something like a setter? Like setVariable?

File: src/ServerBlock.hpp [hunk 5]
```diff
@@ -11,24 +31,45 @@ class ServerBlock
...
+		void setAlias(std::string& str, bool isLocation);
+		void setLocationPath(std::string str);
+		
+		// clear ServerBlock
 		void deleteData();
 	private:
+		Variables _variables;
+		std::vector<Variables> _locations;
+		
+		// TRANSFORMERS
```
Comment (552020) [inline src/ServerBlock.hpp:67]: 🤖

**Review by 552020 — CHANGES_REQUESTED** [review]
Great job! I made some comments in the code ...

File: src/Config.cpp (hunk 1 above)
Comment (dantol29) [reply src/Config.cpp:68]: Sir, yes, Sir!

File: src/Config.cpp (hunk 2 above)
Comment (dantol29) [reply src/Config.cpp:213]: I might need it in the future

File: src/ServerBlock.hpp (hunk 3 above)
Comment (dantol29) [reply src/ServerBlock.hpp:12]: It is hard to watch how variables are being constantly discriminated in this team...
OK!

File: src/ServerBlock.hpp (hunk 4 above)
Comment (dantol29) [reply src/ServerBlock.hpp:34]: It is used only by Config.hpp, so it is should not be public I guess. Maybe there is a way to give access only to a specific class

Comment (552020) [reply src/ServerBlock.hpp:34]: You can use then maybe the friend keyword. https://en.cppreference.com/w/cpp/language/friend

**Comment by dantol29** [comment]
@552020 I pushed with your requested changes

**Review by 552020 — APPROVED** [review]
LGTM!

**Review by lmangall — APPROVED** [review]
LGTM!

**Pull request merged**